Automattic/newspack-plugin
# Dashboard + Checklist + Cohesiveness

### All Submissions:






### Changes proposed in this Pull Request:



Closes #56 .

Sorry for YAGPR (Yet Another Giganto Pull Request 😄). The components are somewhat woven together, and I didn't feel that the architecture could be properly evaluated or communicated if it was split into a bunch of smaller PRs.

This pull request implements dashboard and checklist handling and ties everything together into one (mostly) functional application.

Architecture:
- The Dashboard is a stand-alone area that can render anything into a card (checklists, links, etc.). Under the hood, it's just a 1-screen wizard. See `assets/wizards/dashboard` and `includes/wizards/class-dashboard.php`.
- Checklists are managed and rendered in a centralized way, so we don't need to make a bunch of different independent checklist handlers. Checklists are defined and routed in  `includes/class-checklists.php`. Checklists are rendered in `assets/wizards/checklist`.
- Wizards contain all of the info in their classes (name, description, etc.). They are managed and initialized in `includes/class-wizards.php`. The `Wizards` class provides a central location for retrieving wizards and their info.

Screenshots in details:
<details>

<img width="1449" alt="Screen Shot 2019-06-10 at 11 20 32 AM" src="https://user-images.githubusercontent.com/7317227/59218291-c2ab8a80-8b74-11e9-8e90-57b266323125.png">
<img width="205" alt="Screen Shot 2019-06-10 at 11 20 40 AM" src="https://user-images.githubusercontent.com/7317227/59218292-c2ab8a80-8b74-11e9-947e-64a49b2df311.png">
<img width="1169" alt="Screen Shot 2019-06-10 at 11 21 00 AM" src="https://user-images.githubusercontent.com/7317227/59218294-c3442100-8b74-11e9-8a68-73bfa535468c.png">
<img width="1119" alt="Screen Shot 2019-06-10 at 11 21 14 AM" src="https://user-images.githubusercontent.com/7317227/59218295-c3442100-8b74-11e9-930b-e239d6317acf.png">

</details>

### How to test the changes in this Pull Request:

1. `npm ci; npm run clean; npm run build:webpack`
2. Go to WP Admin > Newspack. Observe most wizards don't have links in the submenu any more.
3. On the dashboard, observe things look good, with a mix of statuses on the cards.
4. Click the "Demo Checklist" or "Memberships" cards. Observe you are taken to the appropriate checklist.
5. Click the "Skip" in the checklist to skip an item or "Do it" to go to the appropriate wizard. Click "I'm done setting up" to return to the dashboard.

The main thing not done in this PR is making the checklist item completion permanent; that'll be done in a future PR.

### Other information:
